src/front/glsl/token.rs

```diff
@@ -38,7 +38,11 @@ pub enum TokenValue {
     Shared,
 
     Restrict,
-    StorageAccess(crate::StorageAccess),
+    /// A `glsl` memory qualifier such as `writeonly`
+    ///
+    /// The associated [`crate::StorageAccess`] is the access being forbidden
+    /// (for example `writeonly` has an associated value of [`crate::StorageAccess::LOAD`])
+    MemoryQualifier(crate::StorageAccess),
```

Comment:
let's use the naming instead of comments. This would look:
```rust
MemoryQualifier {
  forbid: crate::StorageAccess,
}
```
Another choice here, and I think a better one, is to consider `MemoryQualifier(allowed storage access mask)` semantics (i.e. "allow mask" instead of "forbid")

Reply:
I do remember trying to use an allow mask and had some issues, but now I managed to implement it so I don't really remember why.